### PR TITLE
File upload constraints

### PR DESCRIPTION
**Pull request by KorhanOzturk90**
Still wip but early feedback is always appreciated

### CODE REVIEW

File: app/uk/gov/hmrc/fileupload/controllers/envelope.scala [hunk 1]
```diff
@@ -47,21 +48,68 @@ object EnvelopeReport {
...
     )
   }
-
 }
 
 case class CreateEnvelopeRequest(callbackUrl: Option[String] = None,
                                  expiryDate: Option[DateTime] = None,
-                                 metadata: Option[JsObject] = None)
+                                 metadata: Option[JsObject] = None,
+                                 constraints: Option[EnvelopeConstraints] = Some(EnvelopeConstraints(100, 100000000L, 1000000L)))
```
Comment (konradwudkowski) [inline app/uk/gov/hmrc/fileupload/controllers/envelope.scala]: these defaults on line 62 don't match defaults on on lines 79-81

File: app/uk/gov/hmrc/fileupload/write/envelope/Envelope.scala [hunk 2]
```diff
@@ -17,25 +17,32 @@
...
 import cats.data.Xor
+import uk.gov.hmrc.fileupload.controllers.EnvelopeConstraints
 import uk.gov.hmrc.fileupload.write.envelope.Envelope.CanResult
 import uk.gov.hmrc.fileupload.write.infrastructure.{EventData, Handler}
 import uk.gov.hmrc.fileupload.{FileId, FileRefId}
 
 object Envelope extends Handler[EnvelopeCommand, Envelope] {
 
   type CanResult = Xor[EnvelopeCommandNotAccepted, Unit.type]
+  val defaultConstraints = EnvelopeConstraints(100, 25 * 1024 * 1024, 12 * 1024 * 1024)
```
Comment (konradwudkowski) [inline app/uk/gov/hmrc/fileupload/write/envelope/Envelope.scala]: I don't mind if constraints are not read from config although they should be really, but maybe we could define them in one place as part of a static object and read them from there always, this gives one place to change them. This should be a very easy change

File: test/uk/gov/hmrc/fileupload/controllers/EnvelopeConstraintsSpec.scala [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    "should fail the parsing when invalid constraint value is passed" in {
+      val constraintJson =
+        """
+          {
+            "maxItems": 56,
+            "maxSize": "12GB",
+            "maxSizePerItem": "08KB"
+          }
+          """
+      val expectedConstraint = EnvelopeConstraints(100, 25 * 1024 * 1024, 10 * 1024 * 1024)
```
Comment (konradwudkowski) [inline test/uk/gov/hmrc/fileupload/controllers/EnvelopeConstraintsSpec.scala]: this `expectedConstraint` is actually not expected so could be removed

**Comment by konradwudkowski** [comment]
Codewise this looks ok to me apart from 3 minor comments I made.

File: app/uk/gov/hmrc/fileupload/write/envelope/Envelope.scala (hunk 2 above)
Comment (KorhanOzturk90) [inline app/uk/gov/hmrc/fileupload/write/envelope/Envelope.scala]: Done. The default constraints are moved into uk.gov.hmrc.fileupload.read.envelope.Envelope object

File: app/uk/gov/hmrc/fileupload/controllers/envelope.scala (hunk 1 above)
Comment (KorhanOzturk90) [inline app/uk/gov/hmrc/fileupload/controllers/envelope.scala]: They both use the static one from `uk.gov.hmrc.fileupload.read.envelope.Envelope.defaultConstraints` now

**Pull request merged**